src/main/java/at/hannibal2/skyhanni/features/garden/tracking/FarmingTracker.kt

```diff
@@ -0,0 +1,334 @@
+
+    private val patternGroup = RepoPattern.group("garden.tracking")
+
+    /**
+     * REGEX-TEST:  ☘ Cocoa Beans
+     * REGEX-TEST:  ☘ Mushroom
+     */
+    private val tablistUpcomingContestPattern by patternGroup.pattern(
+        "tablist.contest.upcoming",
+        "^ ☘ (?<crop>.+)\$",
+    )
+
+    /**
+     * REGEX-TEST:  ☘ Wheat ◆ Top 0.2%
+     * REGEX-TEST:  ○ Wheat ◆ Top 0.2%
+     */
+    private val tablistActiveContestPattern by patternGroup.pattern(
+        "tablist.contest.active",
+        "^ (?<boost>[☘○]) (?<crop>.+) ◆ Top (?<placement>\\S+)%\$",
+    )
+
+    /**
+     * REGEX-TEST:  Cookie Buff: 28h
+     * REGEX-TEST:  God Potion: 6m
+     */
+    private val tablistEffectsPattern by patternGroup.pattern(
+        "tablist.effects",
+        "^ (?<type>Cookie Buff|God Potion): (?<duration>.+)\$",
+    )
+
+    /**
+     * REGEX-TEST: You have a God Potion active! 32 Minutes
+     */
+    private val tablistFooterGodPotionPattern by patternGroup.pattern(
+        "tablist.footer.godpotion",
+        "You have a God Potion active! (?<length>.+)",
+    )
+
+    @HandleEvent(SecondPassedEvent::class)
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+        if (lastNotification.passedSince() < config.webhook.interval.minutes) return
+
+        status = when {
+            GardenApi.isCurrentlyFarming() -> "Farming"
+            !GardenApi.isCurrentlyFarming() && GardenApi.inGarden() -> "Idle"
+            SkyBlockUtils.inSkyBlock && !GardenApi.inGarden() -> "in Skyblock"
+            SkyBlockUtils.onHypixel -> "Online"
+            else -> status
+        }
+
+        val success = prepareAndSendEmbed(status)
+
+        if (success) lastNotification = SimpleTimeMark.now() else ChatUtils.chat("§cCouldn't send embed (Farming Tracker).")
+    }
+
+    @SubscribeEvent
+    fun onDisconnect(event: FMLNetworkEvent.ClientDisconnectionFromServerEvent) {
+        status = "Offline"
+
+        if (!isEnabled()) return
+
+        val success = prepareAndSendEmbed(status)
+
+        if (success) lastNotification = SimpleTimeMark.now()
+    }
+
+    @HandleEvent
+    fun onWidgetUpdated(event: WidgetUpdateEvent) {
+        if (!isEnabled() && SkyBlockUtils.inSkyBlock) return
+
+        val widget = event.widget
+        val widgetLines = event.widget.lines.map { it.removeColor() }
+        if (widgetLines.isEmpty()) return
+
+        when (widget) {
+            TabWidget.JACOB_CONTEST -> {
+                if (widgetLines[1].contains("Starts In:")) {
+                    tablistUpcomingContestPattern.matchAll(widgetLines) {
+                        getCropEnum(group("crop"))?.let { cropEnum ->
+                            activeAnitaBuff = cropEnum.display + cropEnum.emoji
+                        }
+                    }
+                } else {
+                    tablistActiveContestPattern.matchAll(widgetLines) {
+                        currentPlacement = group("placement").toDouble()
```

Comment:
does not belong in this class. either alr exists in scoreboard/garden/contest api classes, or should belong in contest api if